### PR TITLE
Some TG machine code improvements

### PR DESCRIPTION
**Pull request by yoyobatty**
https://github.com/tgstation/tgstation/pull/46486
https://github.com/tgstation/tgstation/pull/46485
https://github.com/tgstation/tgstation/pull/46502
https://github.com/tgstation/tgstation/pull/46560
https://github.com/tgstation/tgstation/pull/46568
https://github.com/tgstation/tgstation/pull/46570
https://github.com/tgstation/tgstation/pull/45437

Haven't had a chance to fully test this, but it's basically just code improvements and a handful of fixes. Also maintains some of our stuff as well.

Sorry for the dozens of commits, most of this is cherry-picked and webedits.

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -945,22 +951,22 @@
...
-			to_chat(user, "<span class='warning'>There's already a charge hooked up to this door!</span>")
-			return
-		if(detonated)
-			to_chat(user, "<span class='warning'>The maintenance panel is destroyed!</span>")
-			return
-		to_chat(user, "<span class='warning'>You apply [C]. Next time someone opens the door, it will explode.</span>")
-		panel_open = FALSE
-		update_icon()
-		user.transferItemToLoc(C, src, TRUE)
+	else if(istype(C, /obj/item/doorCharge))
```
Comment (ghost) [inline code/game/machinery/doors/airlock.dm:954]: door charges still exist? wtf

**Review by ghost — CHANGES_REQUESTED** [review]
shouldn't you add `spaceship_navigation_beacon.dm` to the hippie dme?

**Comment by yoyobatty** [comment]
> shouldn't you add `spaceship_navigation_beacon.dm` to the hippie dme?

yes

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (yoyobatty) [reply code/game/machinery/doors/airlock.dm:954]: For us, yes.

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
fix the runtimes
```
[00:49:55] Runtime in gravitygenerator.dm,82: Cannot execute null.get status().

  proc name: get status (/obj/machinery/gravity_generator/part/get_status)

  src: the gravitational generator (/obj/machinery/gravity_generator/part)

  src.loc: the floor (112,123,2) (/turf/open/floor/plasteel/dark)

  call stack:

  the gravitational generator (/obj/machinery/gravity_generator/part): get status()

  the gravitational generator (/obj/machinery/gravity_generator/part): update icon()

  the gravitational generator (/obj/machinery/gravity_generator/part): power change()

  the gravitational generator (/obj/machinery/gravity_generator/part): LateInitialize()

  Atoms (/datum/controller/subsystem/atoms): InitAtom(the gravitational generator (/obj/machinery/gravity_generator/part), /list (/list))

  the gravitational generator (/obj/machinery/gravity_generator/part): New(0)

  the gravitational generator (/obj/machinery/gravity_generator/main/station): setup parts()

  the gravitational generator (/obj/machinery/gravity_generator/main/station): Initialize(1)

  Atoms (/datum/controller/subsystem/atoms): InitAtom(the gravitational generator (/obj/machinery/gravity_generator/main/station), /list (/list))

  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null)

  Atoms (/datum/controller/subsystem/atoms): Initialize(29887)

  Master (/datum/controller/master): Initialize(10, 0, 1)
```

**Comment by yoyobatty** [comment]
> fix the runtimes
> 
> ```
> [00:49:55] Runtime in gravitygenerator.dm,82: Cannot execute null.get status().
> 
>   proc name: get status (/obj/machinery/gravity_generator/part/get_status)
> 
>   src: the gravitational generator (/obj/machinery/gravity_generator/part)
> 
>   src.loc: the floor (112,123,2) (/turf/open/floor/plasteel/dark)
> 
>   call stack:
> 
>   the gravitational generator (/obj/machinery/gravity_generator/part): get status()
> 
>   the gravitational generator (/obj/machinery/gravity_generator/part): update icon()
> 
>   the gravitational generator (/obj/machinery/gravity_generator/part): power change()
> 
>   the gravitational generator (/obj/machinery/gravity_generator/part): LateInitialize()
> 
>   Atoms (/datum/controller/subsystem/atoms): InitAtom(the gravitational generator (/obj/machinery/gravity_generator/part), /list (/list))
> 
>   the gravitational generator (/obj/machinery/gravity_generator/part): New(0)
> 
>   the gravitational generator (/obj/machinery/gravity_generator/main/station): setup parts()
> 
>   the gravitational generator (/obj/machinery/gravity_generator/main/station): Initialize(1)
> 
>   Atoms (/datum/controller/subsystem/atoms): InitAtom(the gravitational generator (/obj/machinery/gravity_generator/main/station), /list (/list))
> 
>   Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null)
> 
>   Atoms (/datum/controller/subsystem/atoms): Initialize(29887)
> 
>   Master (/datum/controller/master): Initialize(10, 0, 1)
> ```

This is a TG thing and no, I won't since TG hasn't and I have no idea how to fix them.

**Review by ghost — APPROVED** [review]
apparently tg has the runtime too so /shrug

**Pull request merged**